source/binding/Statements.cpp

```diff
@@ -995,6 +1028,27 @@ bool CaseStatement::verifyConstantImpl(EvalContext& context) const {
     return true;
 }
 
+void CaseStatement::serializeTo(ASTSerializer &serializer) const {
+    serializer.write("condition", toString(condition));
+    serializer.write("check", toString(check));
+    serializer.startArray("items");
+    for (auto const &item: items) {
+        serializer.startArray("expressions");
+        for (auto const &ex: item.expressions) {
+            serializer.serialize(*ex);
+        }
+        serializer.endArray();
+        serializer.startObject("stmt");
```

Comment:
I don't think this will work, it's creating a property inside of an array. Instead I think you want each element in the item array to itself be an object, with one property being the expressions array and the other being the stmt.